pkg/status/status.go

```diff
@@ -119,9 +120,32 @@ type EpochChangeMsg struct {
 }
 
 type NodeBuffer struct {
-	ID             uint64       `json:"id"`
-	Buckets        []NodeBucket `json:"buckets"`
-	LastCheckpoint uint64       `json:"last_checkpoint"`
+	ID         uint64       `json:"id"`
+	Size       int          `json:"size"`
+	Msgs       int          `json:"msgs"`
+	MsgBuffers []*MsgBuffer `json:"msg_buffers"`
+}
+
+type MsgBuffer struct {
+	Component string `json:"component"`
+	Size      int    `json:"size"`
+	Msgs      int    `json:"msgs"`
+}
+
+// Used for sorting MsgBuffers
+// (e.g. to produce deterministic output after iteration over a map).
+// Returns a value > 0 if mb is "greater than" other,
+//                 < 0 if mb is "smaller than" other,
+//                 0 if mb and other are equal.
+// Definition of greater / smaller is arbitrary.
+func (mb *MsgBuffer) Compare(other *MsgBuffer) int {
```

Comment:
Interesting, naively I think I would have just sorted on the component name, but I can see this producing more readable output, nice!